source/core/src/main/com/csse3200/game/entities/factories/PlayerFactory.java

```diff
@@ -73,15 +74,41 @@ public static Entity createPlayer(GdxGame game) {
         player.addComponent((new StatManager()));
         player.addComponent(new LootBoxOverlayComponent());
 
-        PhysicsUtils.setScaledCollider(player, 0.6f, 0.3f);
-        player.getComponent(ColliderComponent.class).setDensity(1.5f);
-        player.getComponent(TextureRenderComponent.class).scaleEntity();
-        player.getComponent(StatManager.class).addStat(new Stat("KangaDefeated", "Kangaroos Defeated", 1));
-        player.getComponent(QuestManager.class).loadQuests();
-
         return player;
     }
 
+    // Create a player NPC to spawn in Combat
+
+    public static Entity createCombatPlayer(String imagePath) {
+        Entity combatPlayer = createCombatPlayerStatic();
+
+        combatPlayer
+                .addComponent(new TextureRenderComponent(imagePath))
+                .addComponent(new CombatStatsComponent(100, 100, 100, 100, 100, 100,true));
+
+        combatPlayer.scaleHeight(90.0f);
+
+        return combatPlayer;
+    }
+
+    /**
+     * Creates a boss NPC to be used as a boss entity by more specific NPC creation methods.
+     *
+     * @return entity
+     */
+    public static Entity createCombatPlayerStatic() {
```

Comment:
should be private (is only used in this class)

Reply:
this is actually used in our CombatArea class to spawn in the player, so can't be private

Reply:
CreateCombatPlayer is used in CombatArea, CreateCombatPlayerStatic is a helper function for CreateCombatPlayer, and so should be private. Albeit after rereading this I'm not even sure why this is 2 functions